apps/condo/domains/property/components/panels/Builder/BuildingPanelView.tsx

```diff
@@ -92,55 +95,53 @@ export const PropertyMapView: React.FC<IPropertyMapViewProps> = ({ Builder, refr
-                                                        if (floorInfo && floorInfo.units.length) {
-                                                            return (
-                                                                <div key={floorInfo.id} style={{ display: 'block' }}>
-                                                                    {
-                                                                        floorInfo.units.map(unit => {
-                                                                            return (
-                                                                                <UnitButton
-                                                                                    key={unit.id}
-                                                                                    noninteractive
-                                                                                >{unit.label}</UnitButton>
-                                                                            )
-                                                                        })
-                                                                    }
-                                                                </div>
-                                                            )
-                                                        } else {
-                                                            return (
-                                                                <EmptyFloor key={`empty_${section.id}_${floorIndex}`} />
-                                                            )
-                                                        }
-                                                    })
-                                                }
-                                                <UnitButton
-                                                    secondary
-                                                    style={{ width: '100%', marginTop: '8px' }}
-                                                    disabled
-                                                >{section.name}</UnitButton>
-                                            </div>
-                                        )
-                                    })
+                                    builder.sections.map(section => (
+                                        <MapSectionContainer
+                                            key={section.id}
+                                            visible={builder.isSectionVisible(section.id)}
+                                        >
+                                            {
+                                                builder.possibleChosenFloors.map(floorIndex => {
+                                                    const floorInfo = section.floors.find(floor => floor.index === floorIndex)
```

Comment:
can we sort `section.floors` once and get data just using index like `section.floors[floorIndex]` ?
searching array this way is quite expensive

other possible solution is to convert `section.floors` to object with keys = floorIndexes

Reply:
@akarjakin  we can't use index here because of `floor.index` is a label, that might be negative

Reply:
@akarjakin Solution with change `sectiion.floors` data type to object is good if not for the need to explicitly define the type of each component. Main problem is that we cannot explicitly describe the data type, because it would be `Record<string, floor>` with an infinite number of keys